internal/repositories/accountMongoRepository.go

```diff
@@ -0,0 +1,149 @@
+package repositories
+
+import (
+	"context"
+	"crud-golang/internal/models"
+	"crud-golang/pkg/client/mongodb"
+	"crud-golang/pkg/logging"
+	"errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"time"
+)
+
+var accountsCollections *mongo.Collection = mongodb.GetCollection(mongodb.DB, "accounts")
+
+type repository struct {
+	accountsCollections *mongo.Collection
+}
+
+func (r repository) Create(user *models.Account) (models.Account, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	_, err := accountsCollections.InsertOne(ctx, user)
+	if err != nil {
+		logging.Error(err)
+		return models.Account{}, errors.New("database error")
+	}
+
+	logging.Infof("Create account by email", user.Email)
+	return *user, err
+}
+
+func (r repository) FindAll() (u []models.Account, err error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	var accounts []models.Account
+	defer cancel()
+
+	results, err := accountsCollections.Find(ctx, bson.M{})
+
+	if err != nil {
+		logging.Fatal(err)
+		return nil, errors.New("database error")
+	}
+
+	defer results.Close(ctx)
+	for results.Next(ctx) {
+		var account models.Account
+		if err = results.Decode(&account); err != nil {
+			logging.Fatal(err)
+			return nil, errors.New("decode error")
+		}
+
+		accounts = append(accounts, account)
+	}
+	logging.Infof("find all accounts")
+	return accounts, err
+}
+
+func (r repository) FindOne(id string) (models.Account, error) {
```

Comment:
Similar comment to the one above for create. Why don't you return a pointer to account. The account empty struct is useless, just consumes memory without any need for it.